src/controller/CHIPDeviceController.cpp

```diff
@@ -1580,10 +1580,88 @@ void DeviceCommissioner::OnDeviceConnectionFailureFn(void * context, PeerId peer
 void DeviceCommissioner::SetupCluster(ClusterBase & base, DeviceProxy * proxy, EndpointId endpoint,
                                       Optional<System::Clock::Timeout> timeout)
 {
-    base.Associate(proxy, 0);
+    base.Associate(proxy, endpoint);
     base.SetCommandTimeout(timeout);
 }
 
+void DescriptorClusterPartsCallback(void * context, const chip::app::DataModel::DecodableList<chip::EndpointId> & data)
+{
+    DeviceCommissioner * commissioner = static_cast<DeviceCommissioner *>(context);
+    EndpointParts parts;
+    auto iter = data.begin();
+    while (iter.Next() && parts.numEndpoints < parts.kMaxEndpoints)
+    {
+        parts.endpoints[parts.numEndpoints++] = iter.GetValue();
+    }
+    CommissioningDelegate::CommissioningReport report;
```

Comment:
Should be checking iter.GetStatus() here?

Reply:
Next checks the status and returns false if the status is an error.

Reply:
Yes, but the point is that we might have gotten here with a malformed (only partially decoded) reply.  If that happens, do we want to ignore the error and press on with what we got, or do something else?  At the very least, we should clearly document the current "ignore that we got a malformed reply, press on" behavior.

Reply:
Added a commit to just pass that back in the report.